packages/playwright-core/src/server/fetch.ts

```diff
@@ -360,15 +360,15 @@ export abstract class APIRequestContext extends SdkObject {
         if (transform) {
           // Brotli and deflate decompressors throw if the input stream is empty.
           const emptyStreamTransform = new SafeEmptyStreamTransform(notifyBodyFinished);
-          body = pipeline(response, emptyStreamTransform, transform, e => {
-            if (e)
-              reject(new Error(`failed to decompress '${encoding}' encoding: ${e}`));
-          });
+          body = response.pipe(emptyStreamTransform).pipe(transform);
         }
 
         body.on('data', chunk => chunks.push(chunk));
         body.on('end', notifyBodyFinished);
-        body.on('error', reject);
+        body.on('error', e => {
+          if (e)
```

Comment:
Let's revert changes in this file as now  we say 'Failed to decompress' on any error even if the response is not compressed.

Does Node 17 remove `pipeline` from its API?